src/webui/www/private/scripts/client.js

```diff
@@ -781,8 +782,38 @@ window.addEvent('load', function() {
         addMainWindowTabsEventListener();
         addSearchPanel();
     }
+
+    registerMagnetHandler();
 });
 
+function registerMagnetHandler() {
+    if (typeof navigator.registerProtocolHandler !== 'function') {
+        return;
+    }
+
+    var hashString = location.hash && location.hash.replace(/^#/, '');
+    var hashParams = hashString && String.parseQueryString(hashString) || {};
+    hashParams.download = '';
+
+    var templateHashString = Object.toQueryString(hashParams);
+    templateHashString = templateHashString.replace('download=', 'download=%s');
+
+    var templateUrl = location.origin + location.pathname
+        + location.search + '#' + templateHashString;
+
+    navigator.registerProtocolHandler('magnet', templateUrl, 'Magnet Handler');
```

Comment:
3rd parameter: `'qBittorrent WebUI magnet handler'` since it will be displayed to the user.